Engine/media/audio/audio.cpp

```diff
@@ -1049,36 +1134,33 @@ void play_new_music(int mnum, SOUNDCLIP *music) {
         return;
     }
 
-    useChannel = prepare_for_new_music ();
-
-    play.cur_music_number=mnum;
+    useChannel = prepare_for_new_music();
+    play.cur_music_number = mnum;
     current_music_type = 0;
-    channels[useChannel] = NULL;
```

Comment:
Are we assuming prepare_for_new_music sets this to null? should we assert?

Reply:
Yes, prepare_for_new_music has this in the end:

<pre>
// Just make sure, because it will be overwritten in a sec
    if (_lock.GetChannel(useChannel) != nullptr)
        stop_and_destroy_channel (useChannel);
</pre>